### ISSUE TITLE
Idea addition and  Guest users

### ISSUE THREAD

**Issue by badrihippo** [abhiniype/approom#1]
I was wondering how we should implement the idea creation form, keeping registered and non-registered users in mind.

What I thought of is this:
- if the user is logged in, then it'll show just the idea creation form, as usual
- if the user is _not_ logged in, then it'll show a small form under the idea form, for the user's name and email address. There will also be sign-up and sign-in links in case the user wants to do that instead.

In the second case, when the form is submitted, a new User record will be created with the given name and email address. A random username will be generated (maybe it can be based on the email?), since it's required by the model. Later, if the users want to sign up, the app can detect from the email address that the user has already made some ideas, and offer to link the accounts.

Any suggestions, or should I go ahead with this idea?

**Comment by abhiniype** [comment]
Seems like a good idea. This also means that there will be two types of accounts: regular account(people who can give ideas, and maybe give ratings and comments on apps) and developer account (people who can upload their own apps as well as give ideas, rate and comment. Is that what you had in mind?

**Comment by badrihippo** [comment]
No, that's actually not what I meant. What I mean was an option for people to add Ideas without having to sign up. So they can just provide an email address, and later if they _do_ decide to sign up, then their older ideas will be already linked to it. As far as the database is concerned, the "guests" will just be deactivated accounts, and can be activated once a password is provided.

But for that we'll probably have to add an email-confirm thing every time and Idea is added/rated to ensure that people don't use invalid (or worse, other peoples') email addresses. So actually now I'm not sure if that's really a good idea.

I don't think we need to have two _different_ types of accounts yet, anyway: non-developers can just not use the developer features.

Maybe we should set up some OpenID/SSO option so that people don't get discouraged by having to fill in another sign-up form?
